ecosystem/cache/handlers.go

```diff
@@ -66,26 +69,36 @@ func (cv *LastestCacheStore) Cost() int64 {
 
 func (s *RelayerCacheServer) GetRelay(ctx context.Context, relayCacheGet *pairingtypes.RelayCacheGet) (cacheReply *pairingtypes.CacheRelayReply, err error) {
 	requestedBlock := relayCacheGet.Request.RequestBlock // save requested block
+	latestKnownBlock := s.getLatestBlock(latestBlockKey(relayCacheGet.ChainID, relayCacheGet.Provider, relayCacheGet.SharedStateId))
+	cacheReply, err = s.getRelayInner(ctx, relayCacheGet, latestKnownBlock)
 
-	cacheReply, err = s.getRelayInner(ctx, relayCacheGet)
 	var hit bool
 	if err != nil {
 		s.cacheMiss(ctx, err)
+		// cacheReply is nil as we didn't get a hit. we still need to set the seen block.
+		if cacheReply == nil {
+			cacheReply = &pairingtypes.CacheRelayReply{}
+		}
 	} else {
 		hit = true
 		s.cacheHit(ctx)
 	}
+
+	// set seen block if required
+	if relayCacheGet.SharedStateId != "" {
+		cacheReply.SeenBlock = latestKnownBlock
+	}
+
 	// add prometheus metrics
 	s.CacheServer.CacheMetrics.AddApiSpecific(requestedBlock, relayCacheGet.ChainID, getMethodFromRequest(relayCacheGet), relayCacheGet.Request.ApiInterface, hit)
 	return
 }
 
-func (s *RelayerCacheServer) getRelayInner(ctx context.Context, relayCacheGet *pairingtypes.RelayCacheGet) (*pairingtypes.CacheRelayReply, error) {
+func (s *RelayerCacheServer) getRelayInner(ctx context.Context, relayCacheGet *pairingtypes.RelayCacheGet, latestKnownBlock int64) (*pairingtypes.CacheRelayReply, error) {
 	inputFormatter, outputFormatter := format.FormatterForRelayRequestAndResponse(relayCacheGet.Request.ApiInterface)
 	relayCacheGet.Request.Data = inputFormatter(relayCacheGet.Request.Data)
 	requestedBlock := relayCacheGet.Request.RequestBlock
-	getLatestBlock := s.getLatestBlock(relayCacheGet.ChainID, relayCacheGet.Provider)
-	relayCacheGet.Request.RequestBlock = lavaprotocol.ReplaceRequestedBlock(requestedBlock, getLatestBlock)
+	relayCacheGet.Request.RequestBlock = lavaprotocol.ReplaceRequestedBlock(requestedBlock, latestKnownBlock)
```

Comment:
let's keep the getLatestBlock original, so the sync and freshness isnt hindered, and separate the seen block and latest block